src/lein_monolith/plugin.clj

```diff
@@ -103,16 +103,38 @@
 (defn build-inherited-profiles
   "Returns a map from profile keys to inherited profile maps."
   [monolith subproject]
-  (reduce
-    (fn [acc [key config]]
-      (let [profile (some-> (choose-inheritance-source monolith config)
-                            (inherited-profile subproject config)
-                            (maybe-mark-leaky config))]
-        (if profile
-          (assoc acc key profile)
-          acc)))
-    nil
-    profile-config))
+  (when (:monolith/inherit subproject)
+    (reduce
+      (fn [acc [key config]]
+        (let [profile (some-> (choose-inheritance-source monolith config)
+                              (inherited-profile subproject config)
+                              (maybe-mark-leaky config))]
+          (if profile
+            (assoc acc key profile)
+            acc)))
+      nil
+      profile-config)))
+
+
+(defn build-dependency-set-profile
+  "Constructs a profile with managed dependencies from the subproject's dependency set.
```

Comment:
Naming/doc: This function returns a map with the profile key, beyond just constructing the profile itself.
```suggestion
(defn build-dependency-profiles
  "Constructs a map with a profile containing managed dependencies from the subproject's chosen dependency set.
```